hugegraph-api/src/main/java/com/baidu/hugegraph/auth/HugeGraphAuthProxy.java

```diff
@@ -1372,8 +1389,21 @@ public void applyStrategies(Admin<?, ?> traversal) {
                 script = translator.translate(traversal.getBytecode());
             }
 
-            verifyNamePermission(HugePermission.EXECUTE,
-                                 ResourceType.GREMLIN, script);
+            /*
+             * Verify gremlin-execute permission for user gremlin(in gremlin-
+             * server-exec worker) and gremlin job(in task worker).
+             * But don't check permission in rest worker, because the following
+             * places need to call traversal():
+             *  1.vertices/edges rest api
+             *  2.oltp rest api (like crosspointpath/neighborrank)
+             *  3.olap rest api (like centrality/lpa/louvain/subgraph)
+             */
+            String caller = Thread.currentThread().getName();
```

Comment:
olap task may run not in grizzly-http-server thread

Reply:
yes olap may run in task worker or olap thread, but currently we can't distinguish gremlin job and olap job